### PR TITLE
[Slim 4] Add PHP 7.0 type definitions

### PR DESCRIPTION
**Pull request by danopz**
According to #2175 I started adding argument  and response type definitions where the type is clear right now. There are still some cases where we can not define some type right now but shouldn't block this one.

~This PR is build upon #2404.~

### CODE REVIEW

File: Slim/RoutingResults.php [hunk 1]
```diff
@@ -49,19 +52,19 @@ class RoutingResults
     /**
      * RoutingResults constructor.
      * @param Dispatcher $dispatcher
-     * @param $httpMethod
-     * @param $uri
+     * @param string $httpMethod
+     * @param string $uri
      * @param int $routeStatus
-     * @param callable|null $routeHandler
+     * @param string|null $routeHandler
```
Comment (akrabat) [inline Slim/RoutingResults.php:58]: I'm not sure that the `$routeHandler` is a string.

Comment (akrabat) [reply Slim/RoutingResults.php:58]: Then again, maybe `$routeHandler` is actually the route's `$identifier`. Can you confirm @l0gicgate ?

Comment (l0gicgate) [reply Slim/RoutingResults.php:58]: `$routeHandler` refers to the route's identifier that's correct. The variable is in fact a string.

Comment (akrabat) [reply Slim/RoutingResults.php:58]: Maybe we should rename the variable?

Comment (l0gicgate) [reply Slim/RoutingResults.php:58]: `$routeIdentifier` sounds pretty descriptive, what do you think?

Comment (danopz) [reply Slim/RoutingResults.php:58]: Should probably done in an extra PR as this one is just for PHP7

Comment (akrabat) [reply Slim/RoutingResults.php:58]: > Should probably done in an extra PR as this one is just for PHP7

Yes, that was my assumption and gentle hint to @l0gicgate  :)

**Pull request merged**